runtime/oti/j9nonbuilder.h

```diff
@@ -1187,6 +1187,28 @@ typedef struct J9SharedDataDescriptor {
 #define J9SHRDATA_USE_READWRITE  8
 #define J9SHRDATA_NOT_INDEXED  16
 #define J9SHRDATA_SINGLE_STORE_FOR_KEY_TYPE  32
+#define J9SHRDATA_SINGLE_STORE_FOR_KEY_TYPE_OVERWRITE  64
+
+typedef struct J9SharedStartupHintsDataDescriptor {
+	UDATA flags;
+	UDATA heapSize1;
+	UDATA heapSize2;
+	UDATA unused1;
+	UDATA unused2;
+	UDATA unused3;
+	UDATA unused4;
+	UDATA unused5;
+	UDATA unused6;
+	UDATA unused7;
+	UDATA unused8;
+	UDATA unused9;
+} J9SharedStartupHintsDataDescriptor;
+
+typedef struct J9SharedLocalStartupHints {
+	U_64 runtimeFlags;
+	char* key;
+	struct J9SharedStartupHintsDataDescriptor* hintsData;
```

Comment:
The structure could be allocated/stored here, rather than using a pointer and allocating the memory at runtime.

Reply:
Fixed